arkane/common.py

```diff
@@ -270,19 +275,33 @@ def load_yaml(self, path, species, pdep=False):
                       'NASA': NASA,
                       'NASAPolynomial': NASAPolynomial,
                       'ThermoData': ThermoData,
+                      'np_array': numpy.array,
                       }
         freq_data = None
         if 'imaginary_frequency' in data:
             freq_data = data['imaginary_frequency']
             del data['imaginary_frequency']
+        if not data['is_ts']:
+            if 'smiles' in data:
+                data['species'] = Species(SMILES=data['smiles'])
+            elif 'adjacency_list' in data:
+                data['species'] = Species().fromAdjacencyList(data['adjacency_list'])
+            elif 'inchi' in data:
+                data['species'] = Species(InChI=data['inchi'])
+            else:
+                raise ValueError('Cannot load ArkaneSpecies from YAML file {0}. Either `smiles`, `adjacency_list`, or '
+                                 'InChI must be specified'.format(path))
+            # Finally, set the species label so that the special attributes are updated properly
+            if label:  # By default use the label passed by the user
```

Comment:
for clarity, maybe relocate the label setting near the `if label != data['label']:` block?

Reply:
Done